module/zfs/zio.c

```diff
@@ -3474,6 +3476,25 @@ zio_vdev_io_done(zio_t *zio)
 	return (ZIO_PIPELINE_CONTINUE);
 }
 
+void
+zio_change_priority(zio_t *pio, zio_priority_t priority)
+{
+	zio_t *cio, *cio_next;
+	zio_link_t *zl = NULL;
+
+	if (pio->io_vd != NULL && pio->io_vd->vdev_ops->vdev_op_leaf &&
```

Comment:
what if the I/O has not been queued yet, how are you changing its priority?

Reply:
Hmmmm. That's a good point. I probably could just change the priority on the zio in this case which should cause any children to inherit the new priority. I'm a bit concerned about locking `zio->io_priority` though. I could probably just take out the `zio->io_lock` for that and for the assignment in `vdev_queue_io()`.  What do you think?

Reply:
This should be fine since we don't really change the priority after `zio_create` (except that vdev_queue_io case which I'm going to nuke).

Reply:
BTW can't we just add something like this:
```
else if (pio->io_priority != priority)
        pio->io_priority = priority;
```

Reply:
@grwilson Would you mind giving an opinion here?

Reply:
I don't think you need to grab the `io_lock` and can just change the priority and doing it in an `else` clause seems simple enough.  By doing this you end up changing the priority as you walk down the children tree.

Reply:
Could that possibly race with `vdev_queue_io()` in such a way that `vdev_queue_io_add()` adds the IO to the wrong tree?

Reply:
I don't think this would be possible for the case since the children will have inherited the parent's priority correctly.